src/main/java/com/google/sps/data/ScheduledCourse.java

```diff
@@ -70,9 +87,17 @@ public ScheduledCourse(Course course, Section lectureSection) {
   public boolean equals(Object other) {
     if (other instanceof ScheduledCourse) {
       ScheduledCourse otherCourse = (ScheduledCourse) other;
```

Comment:
If `other` is `ScheduledCourse` type, why do you cast it? Was this part of the issue you had before with equality?

Reply:
This is standard practice for custom .equals() methods - casting is the only way to access the other object's fields/methods. (If you look through the code at the other equals methods they cast as well)